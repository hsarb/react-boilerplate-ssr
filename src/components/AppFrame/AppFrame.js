// @flow
import * as React from 'react';
import AppBar from 'material-ui/AppBar';
import classnames from 'classnames';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';
import withStyles from 'material-ui/styles/withStyles';

const styles = ({ mixins, palette }: { mixins: Object, palette: Object }) => ({
  root: {
    display: 'flex',
    alignItems: 'stretch',
    minHeight: '100vh',
    width: '100%',
  },
  content: mixins.gutters({
    paddingTop: 80,
    flex: '1 1 100%',
    maxWidth: '100%',
    margin: '0 auto',
  }),
  header: {
    backgroundColor: palette.common.white,
    color: palette.text.primary,
  },
});

type Props = {
  children?: any,
  classes: Object,
  className?: string,
};

function AppFrame(props: Props) {
  const { children, classes, className: classNameProp, ...other } = props;
  const className = classnames(classes.root, classNameProp);

  return (
    <div className={className} {...other}>
      <AppBar className={classes.header}>
        <Toolbar>
          <Typography variant="title" color="inherit">
            React Boilerplate
          </Typography>
        </Toolbar>
      </AppBar>
      <div className={classes.content}>{children}</div>
    </div>
  );
}

export default withStyles(styles)(AppFrame);
