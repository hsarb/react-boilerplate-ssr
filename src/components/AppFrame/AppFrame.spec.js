import AppFrame from './AppFrame';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('AppFrame', () => {
  it('should render correctly', () => {
    const props = {
      children: <div />,
    };

    const wrapper = shallow(<AppFrame {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
