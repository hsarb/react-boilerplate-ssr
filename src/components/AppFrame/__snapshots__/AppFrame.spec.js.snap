// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFrame should render correctly 1`] = `
<AppFrame
  classes={
    Object {
      "content": "AppFrame-content-2",
      "header": "AppFrame-header-3",
      "root": "AppFrame-root-1",
    }
  }
>
  <div />
</AppFrame>
`;
